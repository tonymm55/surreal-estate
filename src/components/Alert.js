import React from "react";

const Alert = ({ message, success }) => {
  return (
    <div className={`alert alert-${success ? "success" : "error"}`}>
      {message}
    </div>
  );
};

export default Alert;
