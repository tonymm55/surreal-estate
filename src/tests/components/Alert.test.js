import React from "react";
import { render } from "@testing-library/react";
import Alert from "../../components/Alert";

describe("Alert Component", () => {
  it("displays an error message", () => {
    const errorMessage = "This is an error message";
    const { getByText, asFragment } = render(<Alert message={errorMessage} type="error" />); // message & type prop
    expect(getByText(/This is an error message/).textContent).toBe(errorMessage);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays a success message", () => {
    const successMessage = "This is a success message";
    const { getByText, asFragment } = render(
      <Alert message={successMessage} type="success" />
    ); // message & type prop
    expect(getByText(/This is a success message/).textContent).toBe(
      successMessage
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays warning message", () => {
    const warningMessage = "Warning, No message!";
    const { getByText, asFragment } = render(
      <Alert message={warningMessage} type="warning" />
    ); // message & type prop
    expect(getByText(/No message!/).textContent).toBe(warningMessage);
    expect(asFragment()).toMatchSnapshot();
  });

  it("does not render an error or a success message if message props is empty", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
  });
});
