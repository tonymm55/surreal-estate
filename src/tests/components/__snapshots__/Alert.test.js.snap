// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component displays a success message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    This is a success message
  </div>
</DocumentFragment>
`;

exports[`Alert Component displays an error message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    This is an error message
  </div>
</DocumentFragment>
`;

exports[`Alert Component displays warning message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Warning, No message!
  </div>
</DocumentFragment>
`;

exports[`Alert Component does not render an error or a success message if message props is empty 1`] = `<DocumentFragment />`;
